unity/Assets/Scripts/AgentManager.cs

```diff
@@ -346,68 +344,190 @@ public BaseFPSAgentController ReturnPrimaryAgent()
+
+        // updates the camera's perspective
+        camera.fieldOfView = fieldOfView;
+        if (orthographic != null) {
+            camera.orthographic = (bool) orthographic;
+            if (orthographic == true && orthographicSize != null) {
+                camera.orthographicSize = (float) orthographicSize;
+            }
+        }
 
-		camera.cullingMask = ~(1 << 11);
+        // supports a solid color skybox, which work well with videos and images (i.e., white/black/orange/blue backgrounds)
+        bool actionFailed = false;
+        if (skyboxColor == "default") {
+            camera.clearFlags = CameraClearFlags.Skybox;
+        } else if (skyboxColor != null) {
+            Color color;
+            bool successfullyParsed = ColorUtility.TryParseHtmlString(skyboxColor, out color);
+            if (successfullyParsed) {
+                camera.clearFlags = CameraClearFlags.SolidColor;
+                camera.backgroundColor = color;
+            } else {
+                actionError(errorMessage: "Invalid skyboxColor! Cannot be parsed as an HTML color.", actionName: actionName);
+                actionFailed = true;
+            }
+        }
 
-		if (this.renderDepthImage || this.renderClassImage || this.renderObjectImage || this.renderNormalsImage || this.renderFlowImage)
-		{
-			gameObject.AddComponent(typeof(ImageSynthesis));
-		}
+        if (!actionFailed) {
+            actionSuccess(actionName: actionName);
+        }
+    }
 
-		this.thirdPartyCameras.Add(camera);
-		gameObject.transform.eulerAngles = action.rotation;
-		gameObject.transform.position = action.position;
+    // Helper used to provide an error message and fail the action in the metadata from the agent manager
+    private void actionError(string errorMessage, string actionName) {
+        // update metadata
+        primaryAgent.errorMessage = errorMessage;
+        primaryAgent.lastAction = actionName;
+        primaryAgent.lastActionSuccess = false;
+        primaryAgent.errorCode = ServerActionErrorCode.Undefined;
 
-		// default to 90 fov on third party camera if value is too small or large
-		camera.fieldOfView = ClampFieldOfView(action.fieldOfView);
+        // end action
+        agentManagerState = AgentState.ActionComplete;
+    }
+
+    // Helper used to update success of agent metadata from the agent manager
+    private void actionSuccess(string actionName) {
+        // update metadata
+        primaryAgent.errorMessage = "";
+        primaryAgent.lastAction = actionName;
+        primaryAgent.lastActionSuccess = true;
+        primaryAgent.errorCode = ServerActionErrorCode.Undefined;
 
-        if (action.orthographic) {
-			// NOTE: orthographicSize uses the original fieldOfView value passed in
-			// TODO: this is an odd overload of the argument, consider adding another parameter
-            camera.orthographicSize = action.fieldOfView;
+        // end action
+        agentManagerState = AgentState.ActionComplete;
+    }
```

Comment:
`actionError` and `actionSuccess` allow the metadata to be updated with the action names (i.e., `AddThirdPartyCamera`, `UpdateThirdPartyCamera`), error message, last action success, and error code.